pinot-core/src/main/java/org/apache/pinot/core/transport/ChannelHandlerFactory.java

```diff
@@ -36,8 +38,9 @@
  * The {@code ChannelHandlerFactory} provides all kinds of Netty ChannelHandlers
  */
 public class ChannelHandlerFactory {
-
   public static final String SSL = "ssl";
+  private static final Map<TlsConfig, SslContext> CLIENT_SSL_CONTEXTS_CACHE = new ConcurrentHashMap<>();
```

Comment:
what about just use hashcode as the key to avoid the object copy?

Reply:
That is a great idea!